docs/infrastructure-gateway-api.md

```diff
@@ -24,84 +24,86 @@ From the gateway api sig:
+    helm upgrade --install nginx-gateway-fabric . --namespace=nginx-gateway -f /opt/genestack/base-helm-configs/nginx-gateway-fabric/helm-overrides.yaml
+    ```
 
-### Controller: Envoyproxy
+    Helm install does not automatically upgrade the crds for this resource. To upgrade the crds you will have to manually install them. Follow the process from :  [Upgrade CRDs](https://docs.nginx.com/nginx-gateway-fabric/installation/installing-ngf/helm/#upgrade-nginx-gateway-fabric-crds)
 
-[Envoyproxy](https://gateway.envoyproxy.io/) is an open-source project that provides an implementation of the Gateway API using Envoyproxy as the data plane.
+=== "Envoyproxy"
 
-#### Installation
+    [Envoyproxy](https://gateway.envoyproxy.io/) is an open-source project that provides an implementation of the Gateway API using Envoyproxy as the data plane.
 
-- Update the `/opt/genestack/base-kustomize/envoyproxy-gateway/base/values.yaml` file according to your requirements.
+    #### Installation
 
-- Apply the configuration using the following command:
+    - Update the `/opt/genestack/base-kustomize/envoyproxy-gateway/base/values.yaml` file according to your requirements.
 
-```shell
-kubectl kustomize --enable-helm /opt/genestack/base-kustomize/envoyproxy-gateway/base | kubectl apply -f -
-```
+    - Apply the configuration using the following command:
 
-After installation, you need to create Gateway and HTTPRoute resources based on your requirements.
+    ```shell
+    kubectl kustomize --enable-helm /opt/genestack/base-kustomize/envoyproxy-gateway/base | kubectl apply -f -
+    ```
 
-### Example to expose an application using Gateway API (Envoyproxy)
+    After installation, you need to create Gateway and HTTPRoute resources based on your requirements.
 
-- In this example, we will demonstrate how to expose an application through a gateway.
+    ### Example to expose an application using Gateway API (Envoyproxy)
 
-- Apply the Kustomize configuration which will create `Gateway` resource:
+    - In this example, we will demonstrate how to expose an application through a gateway.
 
-```shell
-kubectl kustomize /opt/genestack/base-kustomize/gateway/envoyproxy | kubectl apply -f -
-```
+    - Apply the Kustomize configuration which will create `Gateway` resource:
 
-- Once gateway is created, user can expose an application by creating `HTTPRoute` resource.
-  - Sample `HTTPRoute` resource:
-
-  ```shell
-  apiVersion: gateway.networking.k8s.io/v1
-  kind: HTTPRoute
-  metadata:
-    name: test_application
-    namespace: test_app
-  spec:
-    parentRefs:
-    - name: flex-gateway
-      sectionName: http
-      namespace: envoy-gateway-system
-    hostnames:
-    - "test_application.sjc.ohthree.com"
-    rules:
-      - backendRefs:
-        - name: test_application
-          port: 8774
+    ```shell
+    kubectl kustomize /opt/genestack/base-kustomize/gateway/envoyproxy | kubectl apply -f -
     ```
```

Comment:
Can you add the bit about patches here and give example of how the patch should look like and how to apply it so that real usage is covered. For example, what is the format of the patch if we want to do this for SJC or DFW.